zamba/models/config.py

```diff
@@ -0,0 +1,81 @@
+from enum import Enum
+from pathlib import Path
+from typing import Optional, Union
+
+from pydantic import BaseModel as PydanticBaseModel
+from pydantic import DirectoryPath, FilePath
+import yaml
+
+
+class BaseModel(PydanticBaseModel):
+    """Set defaults for all models that inherit from the pydantic base model."""
+
+    class Config:
+        extra = "forbid"
+        use_enum_values = True
+
+
+class ModelClassEnum(str, Enum):
+    cnnensemble = "cnnensemble"
+    custom = "custom"
+    sample = "sample"
+
+
+class ModelLibraryEnum(str, Enum):
+    keras = "keras"
+    pytorch = "pytorch"
+
+
+class ModelProfileEnum(str, Enum):
+    fast = "fast"
+    full = "full"
+
+
+class RegionEnum(str, Enum):
+    us = "us"
+    eu = "eu"
+    asia = "asia"
+
+
+class TrainConfig(BaseModel):
+    train_data: DirectoryPath = None
+    val_data: DirectoryPath = None
+    labels: FilePath = None
+    model_path: FilePath = None
+    model_library: ModelLibraryEnum = "keras"
+    model_class: ModelClassEnum = "custom"
+    tempdir: Optional[Path] = None
+    n_epochs: Optional[int] = 10
+    height: Optional[int] = None
+    width: Optional[int] = None
+    augmentation: Optional[bool] = False
+    early_stopping: Optional[bool] = False
+    save_path: Optional[Path] = None
+    yaml: Optional[FilePath] = None
```

Comment:
What's the use case for this? It's not common to have a data structure know where it is stored. For example, if I change `n_epochs` in code, does that make sure it gets written to the `yaml` file? For that reason, I'd remove this